libcxx/test/support/test_macros.h

```diff
@@ -260,6 +260,10 @@
 
 #define TEST_IGNORE_NODISCARD (void)
 
+#if defined(_LIBCPP_AVAILABILITY_HAS_FROM_CHARS_FLOATING_POINT) && _LIBCPP_AVAILABILITY_HAS_FROM_CHARS_FLOATING_POINT
```

Comment:
```suggestion
#if !(defined(_LIBCPP_VERSION) && !_LIBCPP_AVAILABILITY_HAS_FROM_CHARS_FLOATING_POINT)
```

`_LIBCPP_AVAILABILITY_HAS_FROM_CHARS_FLOATING_POINT` is always defined when we're testing against libc++. Also, if we test the negative form, we will enable the floating point tests on non-libc++ implementations by default, which is what we want.